js/search/domain.js

```diff
@@ -22,7 +20,6 @@ const LINK_TABLE_ID = '4dba9091495e8f277893e0d400e9e092f9f6f551';
  */
 const Domain = new Lang.Class({
     Name: 'Domain',
-    Abstract: true,
```

Comment:
Do you really want to remove the `get_domain_impl()` stuff? Even if we drop the code for Domain v2, doesn't this mean we'd have to separate it all back out again if we wanted Domain v4? Or does Flatpak render that totally unnecessary?